// @unocss/preset-wind4@66.1.1/dist/utils.mjs
import { a as remRE } from './shared/preset-wind4.BSPYtMpU.mjs';
export { C as CONTROL_NO_NEGATIVE, P as PRESET_NAME, S as SpecialColorKey, I as camelize, y as colorCSSGenerator, z as colorResolver, A as colorableShadows, K as compressCSS, c as cornerMap, f as cssMathFnRE, j as cssVarFnRE, F as defineProperty, m as detectThemeValue, d as directionMap, o as directionSize, l as generateThemeVariable, w as getThemeByKey, g as globalKeywords, b as h, h as handler, B as hasParseableColor, J as hyphenate, i as insetMap, G as isCSSMathFn, H as isSize, E as makeGlobalStaticRules, n as numberResolver, q as parseColor, u as parseThemeColor, p as positionMap, r as resolveBreakpoints, D as resolveVerticalBreakpoints, s as splitShorthand, k as themeTracking, t as trackedTheme, v as valueHandlers, e as xyzArray, x as xyzMap } from './shared/preset-wind4.BSPYtMpU.mjs';
export * from '@unocss/rule-utils';
import '@unocss/core';

function createRemToPxProcessor(base = 16) {
  function resolver(utility) {
    if (typeof utility[1] === "string" && remRE.test(utility[1]))
      utility[1] = utility[1].replace(remRE, (_, p1) => `${p1 * base}px`);
  }
  return (utilObjectOrEntry) => {
    if (Array.isArray(utilObjectOrEntry)) {
      resolver(utilObjectOrEntry);
    } else {
      utilObjectOrEntry.entries.forEach(resolver);
    }
  };
}

export { createRemToPxProcessor };
