// hexo-lightning-minify@0.2.2/lib/cacheDB.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheDB = void 0;
const promises_1 = __importDefault(require("node:fs/promises"));
class CacheDB {
    data;
    constructor() {
        this.data = { version: 1 };
    }
    async readDB() {
        await promises_1.default.mkdir('./lightning-minify/images', { recursive: true });
        try {
            const data = await promises_1.default.readFile('./lightning-minify/image.json', 'utf-8');
            this.data = JSON.parse(data);
        }
        catch (error) {
            if (error.code !== 'ENOENT') {
                throw error;
            }
            this.data = { version: 1, images: {} };
        }
    }
    async writeDB() {
        try {
            await promises_1.default.writeFile('./lightning-minify/image.json', JSON.stringify(this.data));
        }
        catch (error) {
            console.error('Error writing image cache db:', error);
        }
    }
    getImageHash(src) {
        if (!this.data.images) {
            this.data.images = {};
        }
        return this.data.images[src]?.hash;
    }
    async setImageHash(src, hash) {
        if (!this.data.images) {
            this.data.images = {};
        }
        this.data.images[src] = { hash };
    }
}
exports.CacheDB = CacheDB;
